src/rgw/rgw_sal_motr.cc

```diff
@@ -2097,7 +2097,11 @@ int MotrObject::MotrDeleteOp::delete_obj(const DoutPrefixProvider* dpp, optional
   rgw_bucket_category_stats& bkt_stat = bkt_header.stats[RGWObjCategory::Main];
   bkt_stat.num_entries--;
   bkt_stat.total_size -= ent.meta.size;
-  bkt_stat.actual_size -= ent.meta.size;
+
+  uint64_t lid = M0_OBJ_LAYOUT_ID(source->mobj->ob_attr.oa_layout_id);
+  unsigned unit_sz = m0_obj_layout_id_to_unit_size(lid);
+  unsigned size_rounded = roundup(ent.meta.size, unit_sz);
```

Comment:
You should use `uint64_t` or `uint32_t` instead of `unsigned int`. 
`uint64_t` is standardized in c++ and guarantees 64bits of storage while size of `unsigned int` depends on underlying platform. I have seen unsigned being used at many places, we need to change it as we go along.

Reply:
Addressed